platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_general.adoc

```diff
@@ -175,3 +175,118 @@ include::sample[dir="snippets/bestPractices/avoidInternal/groovy",files="build.g
 === Tags
 
 <<tags_reference.adoc#tag:upgrades,`#upgrades`>>
+
+[[no_source_in_root]]
+== Do Not Put Source Files in the Root Project
+:keywords: structuring-builds
+
+Do not put source files in your root project; instead, put them in an included (sub)project.
+
+=== Explanation
+
+The root project is a special project in Gradle that serves as the entry point for your build.
+
+It is the place to configure settings that apply globally to the entire build.
+For example, you can declare (but not apply) plugins here to ensure the same plugin version is consistently available across all projects and define other configurations shared by all projects within the build.
+
+It should not be used for source files, instead they should be located in a separate (sub)project.
+
```

Comment:
Do we want to call out that using subproject also makes it easier to add new projects in the future as the whole project doesn't need to be restructured?